src/MultifilesLogger.jl

```diff
@@ -0,0 +1,138 @@
+                                          )
+
+    fileDef2 = FileDefForMultifilesLogger("second.log",
+                                            true, # append
+                                            [(MyModule2,Info)],
+                                            )
+
+    fileDef3 = FileDefForMultifilesLogger("main.log",
+                                            true, # append
+                                            [(Main,Info)],
+                                            )
+
+    # Create the logger and set it as the global logger
+    multifilesLogger =
+        MultifilesLogger([fileDef1,fileDef2,fileDef3])
+
+    global_logger(multifilesLogger)
+
+
+"""
+struct MultifilesLogger <: _iologger
+    filesDefs::Vector{FileDefForMultifilesLogger}
+    logIOs::Dict{Module, Tuple{T,LogLevel}} where T <: IO
+    messageLimits::Dict{Any, Int}
+    flush::Bool
+
+    MultifilesLogger(
+        filesDefs::Vector{FileDefForMultifilesLogger};
+        flush = true) = (x = new(filesDefs,
+                             Dict{Module, Tuple{IO, LogLevel}}(),
+                             Dict{Any, Int}(),
+                             flush);
+                           createIOs!(x);
+                           return x
+                             )
+end
+
+
+
+# CoreLogging.min_enabled_level(logger::MultifilesLogger) = minimum(collect(keys(logger.logPaths)))
+CoreLogging.min_enabled_level(logger::MultifilesLogger) = Info
+
+function createIOs!(logger::MultifilesLogger)
+
+    for fileDef in logger.filesDefs
+
+        # Open one IO per Tuple{Module,LogLevel}
+        # NOTE: there can be several IOs pointing to the same file
+        for t in fileDef.modulesAndLogLevels
+            _module = t[1]
+            _loglevel = t[2]
+            logger.logIOs[_module] =  (open(fileDef.filePath, fileDef.append ? "a" : "w"),
+                                       _loglevel)
+        end
+
+    end
+
+end
+
+function getIOLevelTuple!(logger::MultifilesLogger,
+                         _module::Module,
+                         Wlevel::LogLevel)
+
+    # If a module has no IO we try to find one
+    if !haskey(logger.logIOs,_module)
+        original_module = _module
+        # This stop condition is because 'parentmodule(Main) == Main'
+        while parentmodule(_module) != _module
+            _module = parentmodule(_module)
+
+            # If the parent module has an IO we use it and we update the
+            #   logger.logIOs so that the unreferenced module now points to the
+            #   same IO as the parent module. This is done for performance
+            if haskey(logger.logIOs,_module)
+                # println("Update logger.logIOs")
+                logger.logIOs[original_module] = logger.logIOs[_module]
+                break
+            end
+        end
+
+        # Throw an exception if no logger could be found for the parents
```

Comment:
Are you sure not having an IO available should throw new errors? I'd rather not crash the world because there's no logger specified for `Main`, especially since there doesn't seem to be a default IO (like `devnull`) specified during the creation of the Multifileslogger.

Reply:
OK, not throwing an error anymore